### ISSUE TITLE
Embed assets

### ISSUE THREAD

**Issue by orhun** [theiskaa/markdown2pdf#15]
Would it be possible to embed the fonts under `assets/` into the binary?

I created an AUR package [here](https://aur.archlinux.org/packages/markdown2pdf) but it doesn't currently run because the assets path is hardcoded and they are not a part of the package. We can install them to somewhere in the system path but I think it would be much better if `markdown2pdf` could handle that.

Related error:

```
thread 'main' panicked at src/lib/pdf.rs:108:14:
Failed to load font family: Error { msg: "Failed to open font file assets/fonts/roboto/Roboto-Regular.ttf
", kind: IoError(Os { code: 2, kind: NotFound, message: "No such file or directory" }) }
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by theiskaa** [comment]
@orhun do you have any preferred way, or ideas that could be useful? need something to discuss.  
the only problem i see is that these fonts are hardcoded inside the project, added in `styling.rs`. maybe the user could provide the font name in `markdown2pdfrc.toml` and choose the font.  
e.g.:  
```rust
# level 1 heading style (# heading)
[heading.1]
size = 14
textcolor = { r = 0, g = 0, b = 0 }
afterspacing = 0.5
alignment = "center"
fontfamily = "roboto"
bold = true
italic = false
underline = false
strikethrough = false
backgroundcolor = { r = 255, g = 255, b = 255 }
```

what would be a good way to resolve this without breaking the context?

**Comment by theiskaa** [comment]
For embedding assets, I'm considering the following approach:

```rust
use rust_embed::RustEmbed;
use std::collections::HashMap;
use once_cell::sync::Lazy;

#[derive(RustEmbed)]
#[folder = "assets/"]
pub struct Assets;

// Define available embedded fonts
static EMBEDDED_FONTS: Lazy<HashMap<&str, &str>> = Lazy::new(|| {
    let mut map = HashMap::new();
    map.insert("roboto", "fonts/roboto");
    map.insert("roboto-mono", "fonts/roboto-mono");
    // ...
    map
});
```

The embedded fonts would be used in `styling.rs` like this:

```rust
fn load_embedded_font(font_name: &str) -> Result<FontFamily<FontData>, Error> {
    let base_path = get_embedded_font_path(font_name)
        .ok_or_else(|| Error::new(format!("Font '{}' not found in embedded assets", font_name)))?;
    
    let variants = ["Regular", "Bold", "Italic", "BoldItalic"];
    let mut font_data = Vec::with_capacity(4);

    for variant in variants {
        let path = format!("{}/{}-{}.ttf", base_path, font_name, variant);
        let data = get_font_data(&path)
            .ok_or_else(|| Error::new(format!("Failed to load font variant: {}", path)))?;
        font_data.push(FontData::new(data, None));
    }

    FontFamily::from_bytes(
        font_data[0].clone(),
        font_data[1].clone(),
        font_data[2].clone(),
        font_data[3].clone(),
    )
}
```

However, I'm not entirely satisfied with this approach. While having an embedded font family like "Roboto" as a default is helpful, users should have the flexibility to specify their own fonts, such as by providing a custom path to a font directory. This would allow greater customization while maintaining a fallback to the embedded fonts for cases where no external fonts are provided.

**Comment by orhun** [comment]
You should probably check if there is a custom configuration set and use the embedded fonts as fallback if there isn't. I did something similar in [git-cliff](https://github.com/orhun/git-cliff/blob/main/git-cliff/src/lib.rs#L401) (see the `embed` module)
What do you think?

As for the implementation, `RustEmbed` is the way to go 👍🏼

**Comment by theiskaa** [comment]
I've implemented a demo in #17. If this is not satisfying the requirements i'll check out the git-cliff and make similar tonight.
